features/filesystem/bd/ProfilingBlockDevice.cpp

```diff
@@ -0,0 +1,97 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2017 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "ProfilingBlockDevice.h"
+
+
+ProfilingBlockDevice::ProfilingBlockDevice(BlockDevice *bd, bool debug)
+    : _bd(bd)
+    , _read_count(0)
+    , _program_count(0)
+    , _erase_count(0)
+    , _debug(debug)
+{
+}
+
+int ProfilingBlockDevice::init()
+{
+    return _bd->init();
+}
+
+int ProfilingBlockDevice::deinit()
+{
+    return _bd->deinit();
+}
+
+int ProfilingBlockDevice::read(void *b, bd_addr_t addr, bd_size_t size)
+{
+    _read_count += size;
```

Comment:
Increment of read count should be made only if the read operation was a success.